### PR TITLE
Upgrade Software doc improvements

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/1780

Optimizing some commands and making upgrade paths explicit through a table. I'll add the upgrade path table to other versions of the doc once we agree on the text. See https://astronomer.slack.com/archives/C015V2JFKT5/p1674598093572699

### CODE REVIEW

File: software/upgrade-astronomer.md [hunk 1]
```diff
@@ -170,6 +171,16 @@ This topic contains information about upgrading to specific versions of Astronom
 
 To avoid extended service disruptions, Astronomer recommends upgrading Astronomer Software to a compatible version before you upgrade Kubernetes. To view Astronomer Software and Kubernetes compatibility information, see [Version compatibility reference for Astronomer Software](version-compatibility-reference.md#astronomer-software).
 
+### Supported upgrade paths 
+
+If you're upgrading through multiple Astronomer Software versions in a single upgrade process, review the following table to ensure that you follow the correct upgrade path to your target version. If your combination of **Current version** and **Target version** isn't listed, then it's safe to directly upgrade from your current version to the target version.
```
Comment (kwcanuck) [inline software/upgrade-astronomer.md:176]:
```suggestion
If your upgrade path includes multiple, successive Astronomer Software versions, review the following table to ensure that you're following the correct upgrade path. If your combination of **Current version** and **Target version** isn't listed, you can upgrade directly from your current version to the target version. 
```

**Review by kwcanuck — APPROVED** [review]
The table is a good idea!

File: software/upgrade-astronomer.md [hunk 2]
```diff
@@ -85,18 +85,19 @@ RELEASE_NAME=<astronomer-platform-release-name>
...
 # If you deployed a config change that is intended to reconfigure something inside Airflow,
 # then you may set this value to "true" instead. When it is "true", then each Airflow chart will
 # restart. Note that some stable version upgrades require setting this value to true regardless of your own configuration.
 # If you are currently on Astronomer Software 0.25, 0.26, or 0.27, you must upgrade to version 0.28 before upgrading to 0.29. A direct upgrade to 0.29 from a version lower than 0.28 is not possible.
-helm3 upgrade --namespace $NAMESPACE \
+helm upgrade --namespace $NAMESPACE \
             -f ./config.yaml \
             --reset-values \
             --version $ASTRO_VERSION \
+            --timeout 20m \
```
Comment (jwitz) [inline software/upgrade-astronomer.md:100]:
```suggestion
            -- debug \
            --timeout 20m \
```

File: software_versioned_docs/version-0.28/upgrade-astronomer.md [hunk 3]
```diff
@@ -85,18 +85,19 @@ RELEASE_NAME=<astronomer-platform-release-name>
...
 # upgradeDeployments false ensures that Airflow charts are not upgraded when this script is ran
 # If you deployed a config change that is intended to reconfigure something inside Airflow,
 # then you may set this value to "true" instead. When it is "true", then each Airflow chart will
 # restart. Note that some stable version upgrades require setting this value to true regardless of your own configuration.
 # If you are currently on Astronomer Software 0.25, 0.26, or 0.27, you must upgrade to version 0.28 before upgrading to 0.29. A direct upgrade to 0.29 from a version lower than 0.28 is not possible.
-helm3 upgrade --namespace $NAMESPACE \
+helm upgrade --namespace $NAMESPACE \
             -f ./config.yaml \
             --reset-values \
             --version $ASTRO_VERSION \
```
Comment (jwitz) [inline software_versioned_docs/version-0.28/upgrade-astronomer.md:99]:
```suggestion
            --debug \
            --version $ASTRO_VERSION \
```

File: software_versioned_docs/version-0.29/upgrade-astronomer.md [hunk 4]
```diff
@@ -85,18 +85,19 @@ RELEASE_NAME=<astronomer-platform-release-name>
...
 # upgradeDeployments false ensures that Airflow charts are not upgraded when this script is ran
 # If you deployed a config change that is intended to reconfigure something inside Airflow,
 # then you may set this value to "true" instead. When it is "true", then each Airflow chart will
 # restart. Note that some stable version upgrades require setting this value to true regardless of your own configuration.
 # If you are currently on Astronomer Software 0.25, 0.26, or 0.27, you must upgrade to version 0.28 before upgrading to 0.29. A direct upgrade to 0.29 from a version lower than 0.28 is not possible.
-helm3 upgrade --namespace $NAMESPACE \
+helm upgrade --namespace $NAMESPACE \
             -f ./config.yaml \
             --reset-values \
             --version $ASTRO_VERSION \
```
Comment (jwitz) [inline software_versioned_docs/version-0.29/upgrade-astronomer.md:99]:
```suggestion
            --debug \
            --version $ASTRO_VERSION \
```

File: software_versioned_docs/version-0.30/upgrade-astronomer.md [hunk 5]
```diff
@@ -85,18 +85,19 @@ RELEASE_NAME=<astronomer-platform-release-name>
...
 # upgradeDeployments false ensures that Airflow charts are not upgraded when this script is ran
 # If you deployed a config change that is intended to reconfigure something inside Airflow,
 # then you may set this value to "true" instead. When it is "true", then each Airflow chart will
 # restart. Note that some stable version upgrades require setting this value to true regardless of your own configuration.
 # If you are currently on Astronomer Software 0.25, 0.26, or 0.27, you must upgrade to version 0.28 before upgrading to 0.29. A direct upgrade to 0.29 from a version lower than 0.28 is not possible.
-helm3 upgrade --namespace $NAMESPACE \
+helm upgrade --namespace $NAMESPACE \
             -f ./config.yaml \
             --reset-values \
             --version $ASTRO_VERSION \
```
Comment (jwitz) [inline software_versioned_docs/version-0.30/upgrade-astronomer.md:99]:
```suggestion
            --version $ASTRO_VERSION \
            -- debug \
```

**Pull request merged**